### ISSUE TITLE
[v3] Implement new Concept Exercise: fields

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#844]
This issue describes how to implement the `fields` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Fields in F#.

## Learning objectives

- Know what fields are.
- Know how to define immutable fields.
- Know how to define mutable fields.
- Know what shadowing is.
- Know how to shadow a binding.

## Out of scope

- Static fields.
- Properties.
- Additional constructors.
- `do` bindings.

## Concepts

The Concepts this exercise unlocks are:

- `fields`: know what fields are; know what the difference is between fields and records and when to use which; know what encapsulation is; know how to create an object; 
know how to define methods.
- `shadowing`: know what shadowing is; know how to shadow a binding.

## Prerequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to define a class.
- `mutation`: know how to create mutable bindings.
- `unit`: know about the `unit` type.

Any data types used in this exercise (e.g. `strings` or `integers`) should also be added as prerequisites.

### Hints

- [Fields][fields]: how to define mutable and immutable fields.
- [Shadowing][shadowing]: how to use shadowing to mutate immutable constructor parameters.

### After

- [Fields][fields]: how to define mutable and immutable fields.
- [Shadowing][shadowing]: how to use shadowing to mutate immutable constructor parameters.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/classes.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[fields]: https://fsharpforfunandprofit.com/posts/classes/#private-fields-and-functions-using-let-bindings
[shadowing]: https://fsharpforfunandprofit.com/posts/classes/#mutable-constructor-parameters
